buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt

```diff
@@ -91,6 +92,17 @@ class LifecyclePlugin : Plugin<Project> {
         }
     }
 
+    private
+    fun Project.sharedDependencyAndQualityConfigs() {
```

Comment:
I moved this here to make sure this is done as the first thing and is available for all other plugins/scripts afterwards. Then we do not have to be careful of order in the root build script (apart from applying the lifecycle plugin first).

A follow up should be to get rid of storing dependencies in "extra" altogether.